src/mesh/python/x3d_generator.py

```diff
@@ -51,6 +59,65 @@
 elif args.mesh_type in ['vor_2d_mesh']:
     mesh = mesh_type_dict[args.mesh_type](bnd_per_dim, args.num_cells)
 
+# ------------------------------------------------------------------------------------------------ #
+# -- construct matid and region dictionary if region data arguments are available
+
+# -- set matid data (using region ids and bounds if available)
+matids = np.zeros(mesh.num_cells, dtype=int)
+reg_dict = {}
+reg_ids = []
+if (len(args.reg_ids) > 0):
+    # -- prepend reg_ids and reg_bnd_per_dim with universal region
+    reg_ids = args.reg_ids
+    reg_bnd_per_dim = args.reg_bnd_per_dim
+    reg_ids = [0] + reg_ids
+    reg_bnd_per_dim = args.bnd_per_dim + reg_bnd_per_dim
```

Comment:
Is it worth checking here that a universal region wasn't passed in through the command line args by an unsuspecting user?

Reply:
Good idea! Will add a check for this...

Reply:
This may be okay, since the universal region does not get added if all its cells are covered by 1 or more other regions.  However, removing cells from the universal region (originally at line 111) that overlap with other regions was only being done for the universal region, and not for specified regions, which could cause overlap.  To avoid this, in the recent commit line 111 has been generalized to subtract cells from regions overlapping with regions specified later in the user input list (this has been tested on the 2-region example that was added to the README).